src/Client.php

```diff
@@ -384,9 +448,17 @@ public function sendMessage(Message $message)
    * @param \Maknz\Slack\Message $message
    * @return void
    */
-  public function queueMessage(Message $message)
+  public function queueMessage(Message $message, $numRetries = self::MAX_RETRY_ATTEMPTS)
   {
-    $payload = $this->preparePayload($message);
+    // check for malicious calls and if so, try max times
+    if ($numRetries <= 0)
+    {
+        $numRetries = self::MAX_RETRY_ATTEMPTS;
+    }
```

Comment:
Would it be better and possible to have this logic in `messageHandler` itself? 
Also, we are repeating this block in multiple places.